packages/next-app/src/app/internalmanager/(components)/TokenSelect.tsx

```diff
@@ -156,15 +172,100 @@ function TokenModal({
     }
   }
 
+  const tokenExplorerUrl = buildExplorerTokenURL({
+    chainId: chain!.id,
+    tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
+  });
+
   useEffect(() => {
     if (!internalBalanceData?.user?.userInternalBalances) return;
     getWalletBalance(tokenAdresses as `0x${string}`[]);
   }, [internalBalanceData]);
 
+  useEffect(() => {
+    if (addressRegex.test(tokenSearch)) {
+      setIsTokenSearchDisabled(false);
+      if (
+        tokens.filter((token) => filterTokenInput({ tokenSearch, token }))
+          .length < 1
+      ) {
+        fetchSingleTokenBalance({
+          tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
+        }).then((tokenData) => {
+          setTokens((prev) => [
+            ...prev,
+            {
+              ...tokenData,
+              name: tokenData.symbol,
+              tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
+            },
+          ]);
+        });
+      }
+    } else {
+      setIsTokenSearchDisabled(true);
+    }
+  }, [tokenSearch]);
+
+  function filterTokenInput({
+    tokenSearch,
+    token,
+  }: {
+    tokenSearch: string;
+    token?: TokenItem;
+  }) {
+    {
+      if (tokenSearch === "") return true;
+      const searchString = tokenSearch.toLowerCase();
+      if (token?.name?.toLowerCase().includes(searchString)) return true;
+      if (token?.symbol.toLowerCase().includes(searchString)) return true;
+      if (token?.tokenAddress.toLowerCase().includes(searchString)) return true;
```

Comment:
`#comment` less conventional, but I'd do something in the lines of:

```suggestion
      if (!token) return false;
      const regex = new RegExp(tokenSearch, "i"); // match tokenSearch, case insensitive
      return regex.test(Object.values(token).join(","))
```